packages/ui5-middleware-cfdestination/lib/cfdestination.js

```diff
@@ -25,16 +25,15 @@ module.exports = function ({ resources, options }) {
     const xsappConfig = JSON.parse(fs.readFileSync(options.configuration.xsappJson, 'utf8'));
     const xsappPath = options.configuration.xsappJson.replace("xs-app.json", "");
 
-    // the embedded approuter (the one that runs locally) will ignore all auth mechanisms. 
-    // Even when xsappConfig.authenticationMethod = "route" and only define the routes, 
-    // the approuter will try to read the env var during startup - and fail if they are missing
-    xsappConfig.authenticationMethod = "none";
+    // the default auth mechanism is set to none but the user can pass an auth method using the options
+    xsappConfig.authenticationMethod = options.configuration.authenticationMethod || "none";
```

Comment:
you need to do the optional dance here in order not to have the module crash at runtime:
`(options.configuration && options.configuration.authenticationMethod) || "none"`

Reply:
Is that really needed? As far as i can tell the configuration object is always there. If not it would already crash on line 22. 
I can add it but i dont see that its needed.

Reply:
right on, point taken